File: array/leetcode1.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-
"""
    @Header leetcode1
    @Author fosaken
    @Date   2020/12/19
    @Abstract:
        Address: https://leetcode-cn.com/problems/two-sum/
"""

from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        l = len(nums)
        for i in range(l):
            for j in range(i + 1, l):
                if nums[i] + nums[j] == target:
                    return [i, j]

File: hash/leetcode1.cpp
#include <iostream>
#include <unordered_map>
#include <vector>

using namespace std;

vector<int> twoSum(vector<int>& nums, int target){
    unordered_map<int, int> m;
    for (int i = 0; i < nums.size(); i++){
        if (m.count(nums[i])){
            return {m[nums[i]], i};
        }
        int s = target - nums[i];
        m[s] = i;
    }
    return {};
}

int main(){
    vector<int> nums = {3, 3};
    int target = 6;
    vector<int> re = twoSum(nums, target);
    for (int i = 0; i < re.size(); i++){
        cout << re[i] << ' ';
    }
    system("pause");
    return 0;
}